diff --git a/node_modules/tailwind-rn/dist/tailwind-provider.d.ts b/node_modules/tailwind-rn/dist/tailwind-provider.d.ts
index afa56a7..a423399 100644
--- a/node_modules/tailwind-rn/dist/tailwind-provider.d.ts
+++ b/node_modules/tailwind-rn/dist/tailwind-provider.d.ts
@@ -2,6 +2,7 @@ import * as React from 'react';
 import { ColorSchemeName } from 'react-native';
 import { Utilities } from './types';
 interface Props {
+    children: JSX.Element;
     utilities: Utilities;
     colorScheme?: ColorSchemeName;
 }
