diff --git a/lib/commonjs/assets/index.js b/lib/commonjs/assets/index.js
index 7f1d49228a..c15aca1052 100644
--- a/lib/commonjs/assets/index.js
+++ b/lib/commonjs/assets/index.js
@@ -5,29 +5,6 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.Images = void 0;
 const Images = exports.Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png')
+
 };
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/lib/commonjs/index.js b/lib/commonjs/index.js
index 071b63d05b..ef24460a97 100644
--- a/lib/commonjs/index.js
+++ b/lib/commonjs/index.js
@@ -22,12 +22,7 @@ Object.defineProperty(exports, "EditorHelper", {
     return _EditorHelper.EditorHelper;
   }
 });
-Object.defineProperty(exports, "Images", {
-  enumerable: true,
-  get: function () {
-    return _assets.Images;
-  }
-});
+
 Object.defineProperty(exports, "TenTapView", {
   enumerable: true,
   get: function () {
@@ -341,7 +336,6 @@ Object.keys(_utils).forEach(function (key) {
     }
   });
 });
-var _assets = require("./assets");
 var _EditorHelper = require("./RichText/EditorHelper");
 var _editorHtml = require("./simpleWebEditor/build/editorHtml");
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
diff --git a/lib/module/assets/index.js b/lib/module/assets/index.js
index da8a084ff5..152ab2aff5 100644
--- a/lib/module/assets/index.js
+++ b/lib/module/assets/index.js
@@ -1,27 +1,4 @@
 export const Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png')
+
 };
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/lib/module/index.js b/lib/module/index.js
index e23219ce2e..9b66fc508c 100644
--- a/lib/module/index.js
+++ b/lib/module/index.js
@@ -25,7 +25,6 @@ export * from './bridges/core';
 export * from './bridges/image';
 export * from './types';
 export * from './utils';
-export { Images } from './assets';
 export { EditorHelper } from './RichText/EditorHelper';
 export { editorHtml } from './simpleWebEditor/build/editorHtml';
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/src/assets/index.ts b/src/assets/index.ts
index 858dd108b0..f94d342bf9 100644
--- a/src/assets/index.ts
+++ b/src/assets/index.ts
@@ -1,26 +1,3 @@
 export const Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png'),
+
 };
diff --git a/lib-web/index.mjs b/lib-web/index.mjs
index 8c3dae2c12..bd99f3dcb5 100644
--- a/lib-web/index.mjs
+++ b/lib-web/index.mjs
@@ -13937,21 +13937,8 @@ const eg = se.create({
   addPasteRules() {
     return [
       _e({
-        find: (n, e) => {
-          var t;
-          const r = (t = e == null ? void 0 : e.clipboardData) === null || t === void 0 ? void 0 : t.getData("text/html"), i = [];
-          if (r) {
-            const s = new DOMParser().parseFromString(r, "text/html"), o = s.querySelectorAll("a");
-            o.length && [...o].forEach((l) => i.push({
-              text: l.innerText,
-              data: {
-                href: l.getAttribute("href")
-              },
-              // get the index of the anchor inside the text
-              // and add the length of the anchor text
-              index: s.body.innerText.indexOf(l.innerText) + l.innerText.length
-            }));
-          }
+        find: (n) => {
+          const i = [];
           if (n) {
             const s = ls(n).filter((o) => o.isLink);
             s.length && s.forEach((o) => i.push({
