diff --git a/lib/commonjs/assets/index.js b/lib/commonjs/assets/index.js
index 7f1d49228a8540b3797daee2bcbfc4c88f5b7cbb..c15aca1052f351c1f72bb2e2fc4d5a3871142524 100644
--- a/lib/commonjs/assets/index.js
+++ b/lib/commonjs/assets/index.js
@@ -5,29 +5,6 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.Images = void 0;
 const Images = exports.Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png')
+
 };
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/lib/commonjs/index.js b/lib/commonjs/index.js
index 071b63d05bf746fa4ad0bb4d7996e73e764351aa..ef24460a9788922a7b2a6084643ba39f97e1d1bd 100644
--- a/lib/commonjs/index.js
+++ b/lib/commonjs/index.js
@@ -22,12 +22,7 @@ Object.defineProperty(exports, "EditorHelper", {
     return _EditorHelper.EditorHelper;
   }
 });
-Object.defineProperty(exports, "Images", {
-  enumerable: true,
-  get: function () {
-    return _assets.Images;
-  }
-});
+
 Object.defineProperty(exports, "TenTapView", {
   enumerable: true,
   get: function () {
@@ -341,7 +336,6 @@ Object.keys(_utils).forEach(function (key) {
     }
   });
 });
-var _assets = require("./assets");
 var _EditorHelper = require("./RichText/EditorHelper");
 var _editorHtml = require("./simpleWebEditor/build/editorHtml");
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
diff --git a/lib/module/assets/index.js b/lib/module/assets/index.js
index da8a084ff551d9d3e391f4c8e8b6e376130a4904..152ab2aff5d48645fa67ceb8373bae2645c5b6a6 100644
--- a/lib/module/assets/index.js
+++ b/lib/module/assets/index.js
@@ -1,27 +1,4 @@
 export const Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png')
+
 };
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/lib/module/index.js b/lib/module/index.js
index e23219ce2ee83e8ff94d291e2a9c49aa7deb0d3a..9b66fc508c41c61b55bfd371b26c634b7557bf08 100644
--- a/lib/module/index.js
+++ b/lib/module/index.js
@@ -25,7 +25,6 @@ export * from './bridges/core';
 export * from './bridges/image';
 export * from './types';
 export * from './utils';
-export { Images } from './assets';
 export { EditorHelper } from './RichText/EditorHelper';
 export { editorHtml } from './simpleWebEditor/build/editorHtml';
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/src/assets/index.ts b/src/assets/index.ts
index 858dd108b0e4643b46a94447f37f3a564c2435f4..f94d342bf997c1fb4b78c3eae15fa43319aef497 100644
--- a/src/assets/index.ts
+++ b/src/assets/index.ts
@@ -1,26 +1,3 @@
 export const Images = {
-  indent: require('./indent.png'),
-  outdent: require('./unindent.png'),
-  bold: require('./bold.png'),
-  italic: require('./italic.png'),
-  underline: require('./underline.png'),
-  strikethrough: require('./strikethrough.png'),
-  h1: require('./h1.png'),
-  h2: require('./h2.png'),
-  h3: require('./h3.png'),
-  h4: require('./h4.png'),
-  h5: require('./h5.png'),
-  h6: require('./h6.png'),
-  orderedList: require('./ol.png'),
-  bulletList: require('./ul.png'),
-  checkList: require('./checklist.png'),
-  undo: require('./undo.png'),
-  redo: require('./redo.png'),
-  link: require('./link.png'),
-  palette: require('./palette.png'),
-  code: require('./code.png'),
-  quote: require('./quote.png'),
-  Aa: require('./Aa.png'),
-  a: require('./a.png'),
-  close: require('./close.png'),
+
 };
