diff --git a/ios/MenuViewManager.mm b/ios/MenuViewManager.mm
index a9748300d787222d672b9f9dae33c7d04fd9f791..df4050034ee70b6560a81fbfb1fd4bc47155a9fa 100644
--- a/ios/MenuViewManager.mm
+++ b/ios/MenuViewManager.mm
@@ -7,7 +7,7 @@
 #import "MenuView.h"
 #else
 // OLD ARCH
-#import <react_native_menu-Swift.h>
+#import <react_native_menu/react_native_menu-Swift.h>
 #endif
 
 @interface MenuViewManager : RCTViewManager
