diff --git a/node_modules/@tamagui/vite-plugin/dist/esm/extract.mjs b/node_modules/@tamagui/vite-plugin/dist/esm/extract.mjs
index 2a3131a..ed7b0e7 100644
--- a/node_modules/@tamagui/vite-plugin/dist/esm/extract.mjs
+++ b/node_modules/@tamagui/vite-plugin/dist/esm/extract.mjs
@@ -1,11 +1,11 @@
 import path from "path";
-import { createExtractor, extractToClassNames, getPragmaOptions, loadTamaguiBuildConfigSync } from "@tamagui/static";
+import * as Static from "@tamagui/static";
 import outdent from "outdent";
 import { normalizePath } from "vite";
 const styleUpdateEvent = fileId => `tamagui-style-update:${fileId}`,
   GLOBAL_CSS_VIRTUAL_PATH = "__tamagui_global_css__.css";
 function tamaguiExtractPlugin(optionsIn) {
-  const options = loadTamaguiBuildConfigSync({
+  const options = Static.loadTamaguiBuildConfigSync({
     ...optionsIn,
     platform: "web"
   });
@@ -41,7 +41,7 @@ function tamaguiExtractPlugin(optionsIn) {
       };
     },
     async configResolved(resolvedConfig) {
-      config = resolvedConfig, extractor = createExtractor({
+      config = resolvedConfig, extractor = Static.createExtractor({
         logger: resolvedConfig.logger
       }), shouldReturnCSS = !0, virtualExt = `.tamagui.${shouldReturnCSS ? "css" : "js"}`;
     },
@@ -99,7 +99,7 @@ function tamaguiExtractPlugin(optionsIn) {
           path: validId
         });
       if (shouldDisable) return;
-      const extracted = await extractToClassNames({
+      const extracted = await Static.extractToClassNames({
         extractor,
         source: code,
         sourcePath: validId,
diff --git a/node_modules/@tamagui/vite-plugin/dist/esm/getVitePath.mjs b/node_modules/@tamagui/vite-plugin/dist/esm/getVitePath.mjs
index 51e1599..5141116 100644
--- a/node_modules/@tamagui/vite-plugin/dist/esm/getVitePath.mjs
+++ b/node_modules/@tamagui/vite-plugin/dist/esm/getVitePath.mjs
@@ -1,6 +1,6 @@
 import { join, relative } from "path";
 import resolve from "esm-resolve";
-import { realpath } from "fs-extra";
+import * as fsExtra from "fs-extra";
 async function getVitePath(importer, moduleName, absolute = !1) {
   if (moduleName === "react-native") return "react-native";
   if (moduleName === "react") return "react";
@@ -8,12 +8,21 @@ async function getVitePath(importer, moduleName, absolute = !1) {
   if (moduleName === "react/jsx-dev-runtime") return "react/jsx-dev-runtime";
   if (moduleName[0] === ".") return join("apps/tamastack/src", moduleName) + ".js";
   {
-    const sourceFile = join(process.cwd(), "index.js"),
-      resolved = resolve(sourceFile)(moduleName);
-    if (!resolved) throw new Error("\u274C cant find");
-    const real = await realpath(resolved);
-    let id = real;
-    return absolute || (id = relative(importer, real)), id.endsWith("/react/jsx-dev-runtime.js") && (id = "react/jsx-runtime"), id;
+  const sourceFile = join(process.cwd(), 'index.js')
+  const resolved = resolve(sourceFile)(moduleName)
+  // figure out symlinks
+  if (!resolved) {
+    throw new Error(`❌ cant find`)
+  }
+  const real = await fsExtra.realpath(resolved)
+  let id = real
+  if (!absolute) {
+    id = relative(importer, real)
+  }
+  if (id.endsWith(`/react/jsx-dev-runtime.js`)) {
+    id = 'react/jsx-runtime'
+  }
+  return id
   }
 }
 export { getVitePath };
\ No newline at end of file
diff --git a/node_modules/@tamagui/vite-plugin/dist/esm/plugin.mjs b/node_modules/@tamagui/vite-plugin/dist/esm/plugin.mjs
index 4995003..e16a181 100644
--- a/node_modules/@tamagui/vite-plugin/dist/esm/plugin.mjs
+++ b/node_modules/@tamagui/vite-plugin/dist/esm/plugin.mjs
@@ -1,10 +1,10 @@
-import { watchTamaguiConfig } from "@tamagui/static";
+import { default as Static } from "@tamagui/static";
 import { transformWithEsbuild } from "vite";
 function tamaguiPlugin({
   platform = "web",
   ...options
 }) {
-  const watcher = options.disableWatchTamaguiConfig ? null : watchTamaguiConfig({
+  const watcher = options.disableWatchTamaguiConfig ? null : Static.watchTamaguiConfig({
       platform,
       components: ["tamagui"],
       config: "./src/tamagui.config.ts",
diff --git a/node_modules/@tamagui/vite-plugin/package.json b/node_modules/@tamagui/vite-plugin/package.json
index 13ae8df..382a3cf 100644
--- a/node_modules/@tamagui/vite-plugin/package.json
+++ b/node_modules/@tamagui/vite-plugin/package.json
@@ -1,6 +1,7 @@
 {
   "name": "@tamagui/vite-plugin",
   "version": "1.92.1",
+  "type": "module",
   "types": "./types/index.d.ts",
   "main": "./dist/cjs/index.js",
   "module": "./dist/esm/index.js",
@@ -21,7 +22,7 @@
     ".": {
       "types": "./types/index.d.ts",
       "require": "./dist/cjs/index.js",
-      "import": "./dist/esm/index.js",
+      "import": "./dist/esm/index.mjs",
       "react-native": "./dist/cjs/index.native.js"
     }
   },
