diff --git a/dist/cjs/helpers/propMapper.native.js b/dist/cjs/helpers/propMapper.native.js
index 7bf06919dc353dc83834034a0f8ac4acde30abe5..28662eed7c9d0c7680083e27be2a653432871042 100644
--- a/dist/cjs/helpers/propMapper.native.js
+++ b/dist/cjs/helpers/propMapper.native.js
@@ -59,6 +59,7 @@ var propMapper = function (key, value, styleState, disabled, map) {
     styleProps,
     staticConfig
   } = styleState;
+  if (value === "unset") return;
   var {
     variants
   } = staticConfig;
diff --git a/dist/esm/helpers/propMapper.native.js b/dist/esm/helpers/propMapper.native.js
index 59e8b3bf141f57ffb548a1f6d42a30e052489ff2..65276228095e7ccb6753356a00dfde6f3c29adb3 100644
--- a/dist/esm/helpers/propMapper.native.js
+++ b/dist/esm/helpers/propMapper.native.js
@@ -27,6 +27,7 @@ var propMapper = function (key, value, styleState, disabled, map) {
     styleProps,
     staticConfig
   } = styleState;
+  if (value === "unset") return;
   var {
     variants
   } = staticConfig;
