diff --git a/node_modules/@tiptap/react/dist/index.js b/node_modules/@tiptap/react/dist/index.js
index ca146b7..0b3b451 100644
--- a/node_modules/@tiptap/react/dist/index.js
+++ b/node_modules/@tiptap/react/dist/index.js
@@ -58,7 +58,7 @@ class PureEditorContent extends React.Component {
     }
     init() {
         const { editor } = this.props;
-        if (editor && editor.options.element) {
+        if (editor && !editor.isDestroyed && editor.options.element) {
             if (editor.contentComponent) {
                 return;
             }
