diff --git a/dist/cjs/index.cjs b/dist/cjs/index.cjs
index d3f6a49d994c6a8784e7d2e047d1d0bb511693d8..18ae0b7c7539ff0064110384f7fd43abb3c07b23 100644
--- a/dist/cjs/index.cjs
+++ b/dist/cjs/index.cjs
@@ -450,13 +450,11 @@ class Urbit {
      * be the empty string.
      * @param code The access code for the ship at that address
      */
-    constructor(url, code, desk, fetchFn) {
+    constructor(url, code, desk, fetchFn, urlTransformer = (url) => url) {
         this.url = url;
         this.code = code;
         this.desk = desk;
-        if (isBrowser_1) {
-            window.addEventListener('beforeunload', this.delete);
-        }
+        this.urlTransformer = urlTransformer;
         if (fetchFn) {
             this.fetchFn = fetchFn;
         }
@@ -795,7 +793,7 @@ class Urbit {
         return eventId;
     }
     async sendJSONtoChannel(...json) {
-        const response = await this.fetchFn(this.channelUrl, {
+        const response = await this.fetchFn(this.urlTransformer(this.channelUrl, json), {
             ...this.fetchOptions,
             method: 'PUT',
             body: JSON.stringify(json),
diff --git a/dist/esm/index.js b/dist/esm/index.js
index cd9baa24e3a60f9b29110569b6a02ed1655cac0a..96682a96b3404c7206b36195413d3b3b13bbe92c 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -446,13 +446,11 @@ class Urbit {
      * be the empty string.
      * @param code The access code for the ship at that address
      */
-    constructor(url, code, desk, fetchFn) {
+    constructor(url, code, desk, fetchFn, urlTransformer = (url) => url) {
         this.url = url;
         this.code = code;
         this.desk = desk;
-        if (isBrowser_1) {
-            window.addEventListener('beforeunload', this.delete);
-        }
+        this.urlTransformer = urlTransformer;
         if (fetchFn) {
             this.fetchFn = fetchFn;
         }
@@ -791,7 +789,7 @@ class Urbit {
         return eventId;
     }
     async sendJSONtoChannel(...json) {
-        const response = await this.fetchFn(this.channelUrl, {
+        const response = await this.fetchFn(this.urlTransformer(this.channelUrl, json), {
             ...this.fetchOptions,
             method: 'PUT',
             body: JSON.stringify(json),
