diff --git a/node_modules/@urbit/http-api/dist/esm/index.js b/node_modules/@urbit/http-api/dist/esm/index.js
index 5575d46..9ec251e 100644
--- a/node_modules/@urbit/http-api/dist/esm/index.js
+++ b/node_modules/@urbit/http-api/dist/esm/index.js
@@ -446,10 +446,11 @@ class Urbit {
      * be the empty string.
      * @param code The access code for the ship at that address
      */
-    constructor(url, code, desk, fetchFn) {
+    constructor(url, code, desk, fetchFn, urlTransformer = (url) => url) {
         this.url = url;
         this.code = code;
         this.desk = desk;
+        this.urlTransformer = urlTransformer;
         if (isBrowser_1) {
             window.addEventListener('beforeunload', this.delete);
         }
@@ -784,7 +785,7 @@ class Urbit {
         return eventId;
     }
     async sendJSONtoChannel(...json) {
-        const response = await this.fetchFn(this.channelUrl, {
+        const response = await this.fetchFn(this.urlTransformer(this.channelUrl, json), {
             ...this.fetchOptions,
             method: 'PUT',
             body: JSON.stringify(json),
