diff --git a/dist/ExpensiMark.js b/dist/ExpensiMark.js
index 179639d5a966baaaef8bf215bc552271b88b4306..1df33f7ed24516170da535e0691d545a346ef455 100644
--- a/dist/ExpensiMark.js
+++ b/dist/ExpensiMark.js
@@ -214,6 +214,46 @@ class ExpensiMark {
                 },
                 replacement: '<h1>$1</h1>',
             },
+            {
+                name: 'heading2',
+                process: (textToProcess, replacement, shouldKeepRawInput = false) => {
+                    const regexp = shouldKeepRawInput ? /^## ( *(?! )(?:(?!<pre>|<video>|\n|\r\n).)+)/gm : /^## +(?! )((?:(?!<pre>|<video>|\n|\r\n).)+)/gm;
+                    return this.replaceTextWithExtras(textToProcess, regexp, EXTRAS_DEFAULT, replacement);
+                },
+                replacement: '<h2>$1</h2>',
+            },
+            {
+                name: 'heading3',
+                process: (textToProcess, replacement, shouldKeepRawInput = false) => {
+                    const regexp = shouldKeepRawInput ? /^### ( *(?! )(?:(?!<pre>|<video>|\n|\r\n).)+)/gm : /^### +(?! )((?:(?!<pre>|<video>|\n|\r\n).)+)/gm;
+                    return this.replaceTextWithExtras(textToProcess, regexp, EXTRAS_DEFAULT, replacement);
+                },
+                replacement: '<h3>$1</h3>',
+            },
+            {
+                name: 'heading4',
+                process: (textToProcess, replacement, shouldKeepRawInput = false) => {
+                    const regexp = shouldKeepRawInput ? /^#### ( *(?! )(?:(?!<pre>|<video>|\n|\r\n).)+)/gm : /^#### +(?! )((?:(?!<pre>|<video>|\n|\r\n).)+)/gm;
+                    return this.replaceTextWithExtras(textToProcess, regexp, EXTRAS_DEFAULT, replacement);
+                },
+                replacement: '<h4>$1</h4>',
+            },
+            {
+                name: 'heading5',
+                process: (textToProcess, replacement, shouldKeepRawInput = false) => {
+                    const regexp = shouldKeepRawInput ? /^##### ( *(?! )(?:(?!<pre>|<video>|\n|\r\n).)+)/gm : /^##### +(?! )((?:(?!<pre>|<video>|\n|\r\n).)+)/gm;
+                    return this.replaceTextWithExtras(textToProcess, regexp, EXTRAS_DEFAULT, replacement);
+                },
+                replacement: '<h5>$1</h5>',
+            },
+            {
+                name: 'heading6',
+                process: (textToProcess, replacement, shouldKeepRawInput = false) => {
+                    const regexp = shouldKeepRawInput ? /^###### ( *(?! )(?:(?!<pre>|<video>|\n|\r\n).)+)/gm : /^###### +(?! )((?:(?!<pre>|<video>|\n|\r\n).)+)/gm;
+                    return this.replaceTextWithExtras(textToProcess, regexp, EXTRAS_DEFAULT, replacement);
+                },
+                replacement: '<h6>$1</h6>',
+            },
             /**
              * Converts markdown style images to image tags e.g. ![Expensify](https://www.expensify.com/attachment.png)
              * We need to convert before linking rules since they will not try to create a link from an existing img
