diff --git a/build/ExpoSpeechTranscriberModule.web.js b/build/ExpoSpeechTranscriberModule.web.js
index ef3f500f7b6b0923f9a41d20f715c2d2c2764d04..2e07708b67392b535631699c1abd5cb28bfde6e1 100644
--- a/build/ExpoSpeechTranscriberModule.web.js
+++ b/build/ExpoSpeechTranscriberModule.web.js
@@ -1,6 +1,6 @@
 // import { registerWebModule, NativeModule } from 'expo';
 // import { ExpoSpeechTranscriberModuleEvents } from './ExpoSpeechTranscriber.types';
-export {};
+export default {}; // match default export of native module to avoid static build error
 // class ExpoSpeechTranscriberModule extends NativeModule<ExpoSpeechTranscriberModuleEvents> {
 //   PI = Math.PI;
 //   async setValueAsync(value: string): Promise<void> {
