diff --git a/android/src/main/java/expo/modules/audio/AudioModule.kt b/android/src/main/java/expo/modules/audio/AudioModule.kt
index 3badbf1bff5bba86cf1767d35d77c8c0980bef61..ba280c48dd29515a8d120dfd8e25f89e54663e70 100644
--- a/android/src/main/java/expo/modules/audio/AudioModule.kt
+++ b/android/src/main/java/expo/modules/audio/AudioModule.kt
@@ -266,7 +266,8 @@ class AudioModule : Module() {
             val mediaSource = createMediaItem(source)
             val wasPlaying = ref.player.isPlaying
             mediaSource?.let {
-              ref.player.replaceMediaItem(0, it.mediaItem)
+              // patch: https://github.com/expo/expo/pull/35749
+              ref.setMediaSource(it)
               if (wasPlaying) {
                 ref.player.play()
               }
diff --git a/android/src/main/java/expo/modules/audio/AudioPlayer.kt b/android/src/main/java/expo/modules/audio/AudioPlayer.kt
index ef9ec3c3d8f3bcb008710ad8ad97c7442d1a3738..039a049d4959a77e762d207ca2f9b10a1f27ca63 100644
--- a/android/src/main/java/expo/modules/audio/AudioPlayer.kt
+++ b/android/src/main/java/expo/modules/audio/AudioPlayer.kt
@@ -65,7 +65,8 @@ class AudioPlayer(
     }
   }
 
-  private fun setMediaSource(source: MediaSource) {
+  // patch: https://github.com/expo/expo/pull/35749
+  fun setMediaSource(source: MediaSource) {
     player.setMediaSource(source)
     player.prepare()
     startUpdating()
diff --git a/build/AudioModule.web.js b/build/AudioModule.web.js
index 33bef0f41e5b64340cfcd839db838851c60862b6..350fde1f4c7f564db94af3913cd3c99c3d15ae77 100644
--- a/build/AudioModule.web.js
+++ b/build/AudioModule.web.js
@@ -138,6 +138,13 @@ export class AudioPlayerWeb extends globalThis.expo.SharedObject {
         this.isPlaying = false;
     }
     replace(source) {
+      // patch: match native behavior by sending an unloaded status before replacing the source
+      this.loaded = false;
+      this.emit(PLAYBACK_STATUS_UPDATE, {
+        ...getStatusFromMedia(this.media, this.id),
+        isLoaded: this.loaded,
+      });
+
         this.src = source;
         this.media = this._createMediaElement();
     }
diff --git a/build/ExpoAudio.web.d.ts b/build/ExpoAudio.web.d.ts
index 46dfc0894d05ccb10623e07c8b55d64810482853..032d944779678478f0e2b4d2b3bab2c836b04b9a 100644
--- a/build/ExpoAudio.web.d.ts
+++ b/build/ExpoAudio.web.d.ts
@@ -1,7 +1,12 @@
 import { PermissionResponse } from 'expo-modules-core';
 import { AudioMode, AudioSource, AudioStatus, RecorderState, RecordingOptions, RecordingStatus } from './Audio.types';
-import { AudioPlayer, AudioRecorder, AudioSample } from './AudioModule.types';
 import * as AudioModule from './AudioModule.web';
+
+// patch: https://github.com/expo/expo/pull/37421
+export declare const PLAYBACK_STATUS_UPDATE = "playbackStatusUpdate";
+export declare const AUDIO_SAMPLE_UPDATE = "audioSampleUpdate";
+export declare const RECORDING_STATUS_UPDATE = "recordingStatusUpdate";
+
 export declare function createAudioPlayer(source?: AudioSource | string | number | null, updateInterval?: number): AudioPlayer;
 export declare function useAudioPlayer(source?: AudioSource | string | number | null, updateInterval?: number): AudioModule.AudioPlayerWeb;
 export declare function useAudioPlayerStatus(player: AudioModule.AudioPlayerWeb): AudioStatus;
diff --git a/build/ExpoAudio.web.js b/build/ExpoAudio.web.js
index 74bd2f6deeb673b509009866bc0065d4a86399d0..ce1862be5a3329fd1f0a729916f1016489e6d919 100644
--- a/build/ExpoAudio.web.js
+++ b/build/ExpoAudio.web.js
@@ -1,9 +1,14 @@
 import { useEvent } from 'expo';
 import { useEffect, useState, useMemo } from 'react';
 import * as AudioModule from './AudioModule.web';
-import { AUDIO_SAMPLE_UPDATE, PLAYBACK_STATUS_UPDATE, RECORDING_STATUS_UPDATE } from './ExpoAudio';
 import { createRecordingOptions } from './utils/options';
 import { resolveSource } from './utils/resolveSource';
+
+// patch: https://github.com/expo/expo/pull/37421
+export const PLAYBACK_STATUS_UPDATE = 'playbackStatusUpdate';
+export const AUDIO_SAMPLE_UPDATE = 'audioSampleUpdate';
+export const RECORDING_STATUS_UPDATE = 'recordingStatusUpdate';
+
 export function createAudioPlayer(source = null, updateInterval = 500) {
     const parsedSource = resolveSource(source);
     return new AudioModule.AudioPlayerWeb(parsedSource, updateInterval);
