diff --git a/android/src/main/java/expo/modules/ExpoReactHostFactory.kt b/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
index ed5843d40709d2838e7deb21014383df261864db..05c64e004301f9fad082a3539dcde024f1526b0f 100644
--- a/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
+++ b/android/src/main/java/expo/modules/ExpoReactHostFactory.kt
@@ -72,8 +72,12 @@ object ExpoReactHostFactory {
     override fun getReactNativeConfig(): ReactNativeConfig = reactNativeConfig
 
     override fun handleInstanceException(error: Exception) {
+      val handlers = reactNativeHostWrapper.reactNativeHostHandlers
+      if (handlers.isEmpty()) {
+        throw error
+      }
       val useDeveloperSupport = reactNativeHostWrapper.useDeveloperSupport
-      reactNativeHostWrapper.reactNativeHostHandlers.forEach { handler ->
+      handlers.forEach { handler ->
         handler.onReactInstanceException(useDeveloperSupport, error)
       }
     }
