diff --git a/Libraries/Text/TextInput/Singleline/RCTUITextField.mm b/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
index 03186710893e3629762c295f5e1b8ac51c507fa6..0b570fab2aaea5fd1ef409adb4f3eb8e11689804 100644
--- a/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
+++ b/Libraries/Text/TextInput/Singleline/RCTUITextField.mm
@@ -121,6 +121,8 @@
 {
   NSMutableDictionary<NSAttributedStringKey, id> *textAttributes =
       [_defaultTextAttributes mutableCopy] ?: [NSMutableDictionary new];
+  [textAttributes removeObjectForKey:NSParagraphStyleAttributeName];
+  [textAttributes removeObjectForKey:NSShadowAttributeName];
 
   if (self.placeholderColor) {
     [textAttributes setValue:self.placeholderColor forKey:NSForegroundColorAttributeName];
