diff --git a/ios/RNSScreen.h b/ios/RNSScreen.h
index a536f029eb..6e7f97fc66 100644
--- a/ios/RNSScreen.h
+++ b/ios/RNSScreen.h
@@ -43,6 +43,7 @@ namespace react = facebook::react;
 - (void)setViewToSnapshot;
 - (CGFloat)calculateHeaderHeightIsModal:(BOOL)isModal;
 #endif
+- (BOOL)isRemovedFromParent;
 
 @end
 
diff --git a/ios/RNSScreen.mm b/ios/RNSScreen.mm
index 6d05957932..194f30b2ef 100644
--- a/ios/RNSScreen.mm
+++ b/ios/RNSScreen.mm
@@ -1321,6 +1321,7 @@ Class<RCTComponentViewProtocol> RNSScreenCls(void)
   int _dismissCount;
   BOOL _isSwiping;
   BOOL _shouldNotify;
+  BOOL _isRemovedFromParent;
 }
 
 #pragma mark - Common
@@ -1331,6 +1332,7 @@ Class<RCTComponentViewProtocol> RNSScreenCls(void)
     self.view = view;
     _fakeView = [UIView new];
     _shouldNotify = YES;
+    _isRemovedFromParent = NO;
 #ifdef RCT_NEW_ARCH_ENABLED
     _initialView = (RNSScreenView *)view;
 #endif
@@ -1832,6 +1834,17 @@ Class<RCTComponentViewProtocol> RNSScreenCls(void)
 #endif
 }
 
+- (void)didMoveToParentViewController:(UIViewController *)parent
+{
+  if (parent == nil) {
+    // Since view recycling is disabled, we can rely on a flag indicating that the controller
+    // has been removed from the hierarchy, as it will not be reused.
+    _isRemovedFromParent = YES;
+  }
+
+  [super didMoveToParentViewController:parent];
+}
+
 #ifdef RCT_NEW_ARCH_ENABLED
 #pragma mark - Fabric specific
 
@@ -1875,6 +1888,13 @@ Class<RCTComponentViewProtocol> RNSScreenCls(void)
 }
 #endif
 
+// This method allows us to check whether the Screen has been dismissed;
+// works reliably, because of disabled view recycling.
+- (BOOL)isRemovedFromParent
+{
+  return _isRemovedFromParent;
+}
+
 @end
 
 @implementation RNSScreenManager
diff --git a/ios/RNSScreenStack.mm b/ios/RNSScreenStack.mm
index bfeeb002c6..36c880ddc2 100644
--- a/ios/RNSScreenStack.mm
+++ b/ios/RNSScreenStack.mm
@@ -579,8 +579,26 @@ RNS_IGNORE_SUPER_CALL_END
 
 - (void)setPushViewControllers:(NSArray<UIViewController *> *)controllers
 {
+  /// We collect only active (non-dismissed) screen controllers.
+  /// This is necessary because when a Screen is dismissed natively, its state is sent **asynchronously** to JS.
+  /// As a result, JS might send a delayed update back to native when multiple screens are being dismissed quickly.
+  /// Since view recycling is disabled, once we detect that a screen has been removed from the view hierarchy,
+  /// it won't be reused. This allows us to safely filter out dismissed screens from screens coming from JS state via
+  /// `controllers`.
+  NSMutableArray<UIViewController *> *activeControllers = [NSMutableArray arrayWithCapacity:controllers.count];
+
+  for (UIViewController *vc in controllers) {
+    BOOL isScreen = [vc isKindOfClass:[RNSScreen class]];
+    if (isScreen) {
+      auto screen = (RNSScreen *)vc;
+      if (!screen.isRemovedFromParent) {
+        [activeControllers addObject:vc];
+      }
+    }
+  }
+
   // when there is no change we return immediately
-  if ([_controller.viewControllers isEqualToArray:controllers]) {
+  if ([_controller.viewControllers isEqualToArray:activeControllers]) {
     return;
   }
 
@@ -611,7 +629,7 @@ RNS_IGNORE_SUPER_CALL_END
     return;
   }
 
-  UIViewController *top = controllers.lastObject;
+  UIViewController *top = activeControllers.lastObject;
 #ifdef RCT_NEW_ARCH_ENABLED
   UIViewController *previousTop = _controller.topViewController;
 #else
@@ -626,14 +644,14 @@ RNS_IGNORE_SUPER_CALL_END
 
   if (firstTimePush) {
     // nothing pushed yet
-    [_controller setViewControllers:controllers animated:NO];
+    [_controller setViewControllers:activeControllers animated:NO];
   } else if (top != previousTop) {
-    if (![controllers containsObject:previousTop]) {
+    if (![activeControllers containsObject:previousTop]) {
       // if the previous top screen does not exist anymore and the new top was not on the stack before, probably replace
       // was called, so we check the animation
       if (![_controller.viewControllers containsObject:top] &&
           ((RNSScreenView *)top.view).replaceAnimation == RNSScreenReplaceAnimationPush) {
-        // setting new controllers with animation does `push` animation by default
+        // setting new activeControllers with animation does `push` animation by default
 #ifdef RCT_NEW_ARCH_ENABLED
         // This is a workaround for the case, when in the app we're trying to do `replace` action on screens, when
         // there's already ongoing transition to some screen. In such case, we're making the snapshot, but we're trying
@@ -641,15 +659,15 @@ RNS_IGNORE_SUPER_CALL_END
         // _UIParallaxDimmingView instead). At the moment of RN 0.74 we can't queue the unmounts for such situation
         // either, so we need to turn off animations, when the view is not yet mounted, but it will appear after the
         // transition of previous replacement.
-        [_controller setViewControllers:controllers animated:previousTop.view.window != nil];
+        [_controller setViewControllers:activeControllers animated:previousTop.view.window != nil];
 #else
-        [_controller setViewControllers:controllers animated:YES];
+        [_controller setViewControllers:activeControllers animated:YES];
 #endif // RCT_NEW_ARCH_ENABLED
       } else {
         // last top controller is no longer on stack
-        // in this case we set the controllers stack to the new list with
+        // in this case we set the activeControllers stack to the new list with
         // added the last top element to it and perform (animated) pop
-        NSMutableArray *newControllers = [NSMutableArray arrayWithArray:controllers];
+        NSMutableArray *newControllers = [NSMutableArray arrayWithArray:activeControllers];
         [newControllers addObject:previousTop];
         [_controller setViewControllers:newControllers animated:NO];
         [_controller popViewControllerAnimated:YES];
@@ -658,7 +676,7 @@ RNS_IGNORE_SUPER_CALL_END
       // new top controller is not on the stack
       // in such case we update the stack except from the last element with
       // no animation and do animated push of the last item
-      NSMutableArray *newControllers = [NSMutableArray arrayWithArray:controllers];
+      NSMutableArray *newControllers = [NSMutableArray arrayWithArray:activeControllers];
       [newControllers removeLastObject];
 
       [_controller setViewControllers:newControllers animated:NO];
@@ -666,11 +684,11 @@ RNS_IGNORE_SUPER_CALL_END
     } else {
       // don't really know what this case could be, but may need to handle it
       // somehow
-      [_controller setViewControllers:controllers animated:NO];
+      [_controller setViewControllers:activeControllers animated:NO];
     }
   } else {
     // change wasn't on the top of the stack. We don't need animation.
-    [_controller setViewControllers:controllers animated:NO];
+    [_controller setViewControllers:activeControllers animated:NO];
   }
 }
 
