diff --git a/src/useLocalStorage/useLocalStorage.ts b/src/useLocalStorage/useLocalStorage.ts
index 7b230766a69e9e85f75aa93534a8ceb158007557..5cfae88ded030b66207ee5923e5e56c4823d0c80 100644
--- a/src/useLocalStorage/useLocalStorage.ts
+++ b/src/useLocalStorage/useLocalStorage.ts
@@ -64,6 +64,11 @@ function useLocalStorage<T>(key: string, initialValue: T): [T, SetValue<T>] {
       window.dispatchEvent(new Event('local-storage'))
     } catch (error) {
       console.warn(`Error setting localStorage key “${key}”:`, error)
+
+      // Allow value to be a function so we have the same API as useState
+      const newValue = value instanceof Function ? value(storedValue) : value
+      // Save state
+      setStoredValue(newValue)
     }
   })
 
