--- a/sur/aquarium.hoon	2026-01-16 17:07:56
+++ b/sur/aquarium.hoon	2026-04-08 22:28:50
@@ -83,7 +83,7 @@
       [%init ~]
       [%request id=@ud request=request:http]
       [%turf p=(list turf)]
-      [%unto =unto:gall]  ::  internal gifts to host ships
-                          ::   (%facts are transformed to %raw-facts)
+      [%unto p=unto:gall]
+      [%response =http-event:http]
   ==
 --
